test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -0,0 +1,225 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+public class CompatibilityChecker {
+    private static final int READER_TIMEOUT_MS = 2000;
+    public  URI controllerURI;
+    public  StreamManager streamManager;
+    public StreamConfiguration streamConfig;
+
+    public  void setUp() {
+        controllerURI = URI.create("tcp://localhost:9090");
+        streamManager = StreamManager.create(controllerURI);
+        streamConfig = StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(1)).build();
+    }
+
+    /**
+    * This method is Checking the working of the read and write event of the stream.
+    * Here we are trying to create a stream and a scope, and then we are writing a couple of events.
+    * And then reading those written event from the stream.
+    */
+    public void checkWriteAndReadEvent() {
+        String scopeName = "write-and-read-test-scope";
+        String streamName = "write-and-read-test-stream";
+        streamManager.createScope(scopeName);
+        streamManager.createStream(scopeName, streamName, streamConfig);
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(), EventWriterConfig.builder().build());
+        String readerGroupId = UUID.randomUUID().toString().replace("-", "");
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder().stream(Stream.of(scopeName, streamName)).build();
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scopeName, controllerURI);
+        readerGroupManager.createReaderGroup(readerGroupId, readerGroupConfig);
+        @Cleanup
+        EventStreamReader<String> reader =  clientFactory.createReader("reader-1", readerGroupId, new UTF8StringSerializer(), ReaderConfig.builder().build());
+        // Writing 10 Events to the stream
+        for (int event = 0; event < 10; event++) {
+            writer.writeEvent("event test" + event);
+        }
+        log.info("Reading all the events from {}, {}", scopeName, streamName);
+        EventRead<String> event = null;
+        // Reading all those 10 events from the stream
+        do {
+            try {
+                event = reader.readNextEvent(READER_TIMEOUT_MS);
+                if (event.getEvent() != null) {
+                    System.out.format("Read event '%s'%n", event.getEvent());
+                }
+            } catch (ReinitializationRequiredException e) {
+                //There are certain circumstances where the reader needs to be reinitialized
+                e.printStackTrace();
+            }
+        } while (event.getEvent() != null);
+        log.info("No more events from {}, {}", scopeName, streamName);
```

Comment:
This is not asserting that it is reading back the 10 expected events. So, a system may not return them and the execution of this will succeed anyway.

Reply:
You are not doing what I meant: If you know the events being written and you know how many should be there, just assert their content and their count.

Reply:
done.